src/engine/database/neo4j.rs

```diff
@@ -805,3 +837,56 @@ impl From<Value> for bolt_proto::value::Value {
         }
     }
 }
+
+impl<GlobalCtx, RequestCtx> TryFrom<bolt_proto::value::Value> for Node<GlobalCtx, RequestCtx>
+where
+    GlobalCtx: GlobalContext,
+    RequestCtx: RequestContext,
+{
+    type Error = crate::Error;
+
+    fn try_from(value: bolt_proto::value::Value) -> Result<Self, Error> {
+        match value {
+            bolt_proto::value::Value::Node(n) => {
+                let type_name = &n.labels()[0];
+                let properties: &HashMap<String, bolt_proto::value::Value> = &n.properties();
+                let props_value = Value::try_from(properties.clone())?;
+                let props = match HashMap::<String, Value>::try_from(props_value) {
+                    Ok(v) => v,
+                    Err(_) => {
+                        return Err(Error::TypeConversionFailed {
+                            src: "Value".to_string(),
+                            dst: "HashMap::<String, Value>".to_string(),
+                        })
+                    }
+                };
+                Ok(Node::new(type_name.to_string(), props))
+            }
+            _ => Err(Error::TypeConversionFailed {
+                src: "bolt_proto::value::Value".to_string(),
+                dst: "Node".to_string(),
+            }),
+        }
+    }
+}
+
+impl TryFrom<HashMap<String, bolt_proto::value::Value>> for Value {
+    type Error = Error;
+
+    fn try_from(hm: HashMap<String, bolt_proto::value::Value>) -> Result<Value, Error> {
+        let mut hmv = HashMap::<String, Value>::new();
+        for (key, bolt_value) in hm.iter() {
+            let value = match Value::try_from(bolt_value.clone()) {
+                Ok(v) => v,
+                Err(_) => {
+                    return Err(Error::TypeConversionFailed {
+                        src: "Value".to_string(),
```

Comment:
I could use some help here. If I try `src: format!("{:#?}", bolt_value)`, I get a "value borrowed after move". My naive fix is to clone `bolt_value` when I pass it to `Value::try_from` above, but we are trying to avoid cloning. Any ideas?

Reply:
Hmmm... what about `format!("{:#?}", &bolt_value)`?  If I recall correctly, the `format!` macro should be able to work off a reference. I don't think it needs ownership.

Reply:
Using `if let` still doesn't fix the move after borrow issue.

Reply:
That confuses me.  Maybe we can set up a screen share at you convenience?